### PR TITLE
Add note to clarify that variants can be used to specify multiple usecases in NXmx's NXbeam

### LINKED ISSUE #766: Add note re: variants to NXmx's NXbeam
Add note to NeXus [variants](https://manual.nexusformat.org/datarules.html#variants) to NXmx's NXbeam to clarify how to support multiple use cases at once, such as a calibrated wavelength from 1D spectrum.

### PR DESCRIPTION
**Pull request by phyy-nx**
Closes #766.

### CODE REVIEW

**Comment by phyy-nx** [comment]
Is there some markup I could use to link directly to https://manual.nexusformat.org/datarules.html#variants?

Thanks.

**Comment by prjemian** [comment]
You can use this to link to the [Variants](https://manual.nexusformat.org/datarules.html#variants?) section:

```
:ref:`Design-Variants`
```

this will use the section title as text.  If you want override that default, use:

```
:ref:`my custom text <Design-Variants>`
```

File: applications/NXmx.nxdl.xml [hunk 1]
```diff
@@ -605,6 +605,11 @@
 							of dimensions **np** by **m** (slow to fast) with the
 							relative weights in incident_wavelength_weight as a 2D
 							array.
+
+						        Note, variants are a good way to represent several of
+							these usecases in a single dataset, e.g. if a calibrated,
```
Comment (prjemian) [inline applications/NXmx.nxdl.xml:610]: two words: `use cases`

File: applications/NXmx.nxdl.xml [hunk 2]
```diff
@@ -605,6 +605,11 @@
...
+							single-value wavelength value is available along with the
+							original spectrum it was calibrated from.
```
Comment (prjemian) [inline applications/NXmx.nxdl.xml:612]: ---: `it was calibrated from.`
+++: `from which it was calibrated.`

File: applications/NXmx.nxdl.xml (hunk 1 above)
Comment (prjemian) [reply applications/NXmx.nxdl.xml:610]: Also, the indentation here is not consistent with the previous lines.  This is the reason for the failure on [travis-ci](https://travis-ci.org/github/nexusformat/definitions/jobs/675042570#L967-L986).  Copy the leading white space from the previous line verbatim and use as the indentation for each of the following lines in this part.

**Comment by phyy-nx** [comment]
Thanks, all comments addressed and variants link added.

**Comment by prjemian** [comment]
Still an error to resolve as identified by [travis-ci](https://travis-ci.org/github/nexusformat/definitions/jobs/675380405#L388)

    lxml.etree.XMLSyntaxError: Opening and ending tag mismatch: Design-Variants line 609 and doc, line 614, column 13

**Comment by prjemian** [comment]
Might be tough to resolve this one.  We need the `<` and `>` signs for the Sphinx/reST markup but XML has problems with them as content.  Encode them using [XML entities](https://www.w3resource.com/xml/entities.php):

character | XML entity
--- | ---
`<` | `&lt;`
`>` | `&gt;`

* https://www.tutorialspoint.com/xml/xml_character_entities.htm
* https://en.wikipedia.org/wiki/List_of_XML_and_HTML_character_entity_references

**Comment by prjemian** [comment]
The HTML entity is used to encode the reST structure we need *within* an XML element.  Thus, the structure **must** conform to the XML rules which have a very clear definition of the `<` and `>` characters.  The entity is used to encode those when they are not used as part of the structure but rather the content of the XML element.  The software that reads the NXDL files and prepares the manual will take of converting the entities into the proper reST code.

**Comment by phyy-nx** [comment]
Thanks. Bugfixed and it looks like the Travis build finished.

**Comment by phyy-nx** [comment]
@prjemian ok to merge?

**Pull request merged**